docs/articles/Vector-Indexes.md

```diff
@@ -0,0 +1,385 @@
+
+1. Divide the vector space into clusters.
+2. Break vectors within the cluster into smaller chunks. For example, if the original vector consists of 12 elements, it can be broken into 3 chunks, each with 4 elements.
+3. Quantize the chunks (4-dimensional vectors) into bits using product quantization.
+
+At search time, the same algorithm is applied to the query vector. It is first associated with a cluster by calculating the distance between the query and the cluster centroids. Then, it is broken into chunks and quantized. The encoded sub-vectors are compared with the encoded data within the cluster using distance metrics. This storage method has various benefits, such as it saves on space, and since we’re comparing smaller vectors, it improves vector search times.
+
+#### IVFSQ
+
+Inverted File Scalar Quantization (IVFSQ) is similar to IVFPQ but uses a simpler quantization algorithm. It begins by clustering the data points and then quantizing each vector separately. The quantization algorithm quantizes each vector dimension. The first step is to create bins to map the vector elements to. Each element is mapped to a bin, and depending on the bin range, the floating point number is converted to a scalar integer.
+
+Suppose we have a vector x=[1.2,3.5,−0.7,2.1]. As it has 4 dimensions, we will define 4 quantization bins such as the following:
+
+- Bin 0: [−1.0,0.0)[-1.0,0.0)[−1.0,0.0)
+- Bin 1: [0.0,1.0)[0.0,1.0)[0.0,1.0)
+- Bin 2: [1.0,2.0)[1.0,2.0)[1.0,2.0)
+- Bin 3: [2.0,4.0)[2.0,4.0)[2.0,4.0)
+
+Each vector element will be distributed into a bin as follows:
+
+- For x1=1.2: It falls into Bin 2 [1.0,2.0), so it is quantized to 2.
+- For x2=3.5: It falls into Bin 3 [2.0,4.0), so it is quantized to 3.
+- For x3=−0.7: It falls into Bin 0 [−1.0,0.0), so it is quantized to 0.
+- For x4=2.1: It falls into Bin 3 [2.0,4.0), so it is quantized to 3.
+
+The final quantized vector becomes [2,3,0,3].
+
+### DiskANN
+
+Most ANN algorithms are designed for in-memory computation, which can be a bottleneck when working with big data. [DiskANN](https://suhasjs.github.io/files/diskann_neurips19.pdf) is built to leverage the large memory and high-speed capabilities of Solid-State Drives (SSDs). It uses a graph-based structure to index the vector data points. The graph nodes represent individual vectors and are connected to other nodes based on their similarity. For this implementation, the authors introduce Vamana, a special graph structure with a smaller "diameter" (the maximum number of hops needed to reach any two points), minimizing the number of sequential disk reads required during the search.
+
+The graph index is entirely stored on disk and accessed during search. The overall approach is cost-effective and scalable and implements an efficient search strategy.
+
+### Scalable Progressive Approximate Nearest Neighbor Search (SPANN)
```

Comment:
I can't find this acronym spelled out anywhere in the paper or elsewhere. The authors refer to is as Highly-efficient Billion-scale Approximate Nearest Neighbor Search. Do you have a reference for this?

Reply:
the linked repo mentions SPTAG - space partition tree and graph which SPANN is based on. So probably I'd expand this acronym as "SPTAG-based Approximate Nearest Neighbor Search"

Reply:
Repo: https://github.com/microsoft/SPTAG